### ISSUE TITLE
Unable to delete objets in the admin panel because id is not an integer

### ISSUE THREAD

**Issue by Aqueuse** [nesdis/djongo#558]
Hello,

Sorry if it was asked but I can't delete my objects in the admin panel, it's throwing me this error : 

http://127.0.0.1:8000/admin/ours_agile/projet/
3.2.4
ValueError
Field 'id' expected a number but got 'None'.

Apparently, djongo use the ObjectID type of mongoDB as an id without translating it to an integer.

How to solve this issue ? I want to use the admin page of Django, I love it.

**Comment by shreyanshu7101904** [comment]
@Aqueuse could you please share your models on which you are doing this operation?

**Comment by Aqueuse** [comment]
Yes, I can, but I don't include anything special in my model, I juste want to use the _id created by mongoDB as the primary key, like I would do with sqllite3 for example.

`class Projet(models.Model):
    title = models.CharField(max_length=100)
    html_content = models.TextField(blank=True)

    def __str__(self):
        return f"{self.title}"`

**Comment by rudolfce** [comment]
Hello, @Aqueuse!

Could you please check if the solutions from #557 solve your problem? This looks like you didn't run the initial migrations for your model, or you're using django's default `models` module.

Please, let us know if this doesn't help

**Comment by shreyanshu7101904** [comment]
Try to create your models like that

```
from boson import ObjectId
from djongo import models
Class category(models.Model):
    id = ObjectIdField(db_column="_id", primary_key=True)
    category = models.Charfield(max_length=40)
```

**Comment by Aqueuse** [comment]
> Try to create your models like that
> 
> ```
> from boson import ObjectId
> from djongo import models
> Class category(models.Model):
>     id = ObjectIdField(db_column="_id", primary_key=True)
>     category = models.Charfield(max_length=40)
> ```

yes, that's was I finished to do, but in sqlite3 I don't need to specify the id, it's automatic and the admin page works perfectly fine. Could it be possible to enhance djongo to create a """good""" id that the admin page could use, like an integer ?

**Comment by Aqueuse** [comment]
> Hello, @Aqueuse!
> 
> Could you please check if the solutions from #557 solve your problem? This looks like you didn't run the initial migrations for your model, or you're using django's default `models` module.
> 
> Please, let us know if this doesn't help

Hello,

As a student in Django I follow the tutorial, so I always use makemigrations then migrate when I change my models.

I have tried to use directly djongo.models (as in https://github.com/nesdis/djongo/issues/557) in my class declaration, but it don't change anything (can't delete from the admin page, produce the same error => Field 'id' expected a number but got 'None'.).

It create a collection who look like that in mongodb : 

{
    "_id": {
        "$oid": "60d1db2cd71748591bb91bc4"
    },
    "title": "test",
    "html_preview": "test"
}

From my beginner perspective, It seems that the admin page should use $oid instead of _id.

This is the code I produced to get to this result :   

```
  from djongo import models as djongomodels

  class Test(djongomodels.Model):
      title = models.CharField(max_length=100)
      html_preview = models.TextField(blank=True)

      def __str__(self):
          return f"{self.title}"
```

**Comment by rudolfce** [comment]
When you generated and ran the migration, you were using the `djongo.models` or `django.db.models`? The migration is necessary with the adequate models to create the `__schema__` collection, that generates the auto id.

Also, the best practice for the current state of the project is to use `djongo.models` to completely replace the `django.db.models`, so you don't need to keep the django models module for the field declarations.

**Comment by Aqueuse** [comment]
Yes, I have do all the migration after switching (I would not have been able to access it from the admin page otherwise).

**Comment by shreyanshu7101904** [comment]
> > Try to create your models like that
> 
> > 
> 
> > ```
> 
> > from boson import ObjectId
> 
> > from djongo import models
> 
> > Class category(models.Model):
> 
> >     id = ObjectIdField(db_column="_id", primary_key=True)
> 
> >     category = models.Charfield(max_length=40)
> 
> > ```
> 
> 
> 
> yes, that's was I finished to do, but in sqlite3 I don't need to specify the id, it's automatic and the admin page works perfectly fine. Could it be possible to enhance djongo to create a """good""" id that the admin page could use, like an integer ?

I believe for older versions of django i.e < 3 this package was working fine I also did some changes to work it properly. You need to play around bison fields and Dijon go fields get your work done

**Comment by dsnikam** [comment]
> Try to create your models like that
> 
> ```
> from boson import ObjectId
> from djongo import models
> Class category(models.Model):
>     id = ObjectIdField(db_column="_id", primary_key=True)
>     category = models.Charfield(max_length=40)
> ```

This worked for me. I was using django.db.models
